### ISSUE TITLE
Faux-duplicate IPs across VLANs

### ISSUE THREAD

**Issue by Azmodeszer** [bb-Ricardo/netbox-sync#344]
I'm running into a primary IP assignment warning:

`WARNING: Current interface 'vNIC 1 (Office) (<SERVER1>)' for IP '1.2.3.4/21' and this one 'vNIC 1 (Virtual LAB) (<SERVER2>)' are both enabled. IP assignment skipped because it is unclear which one is the correct one!`

Okay, fair enough, but the thing here is that that's two different VLANs, one used to simulate the other and exactly map it. So the IPs being the same is as it should be. The sync already imports the VLANs for each device's interface correctly, so is there a way to tell it to ignore the warning and set the IPs if the VLANs differ?

**Comment by Azmodeszer** [comment]
Related to this, I'm also getting a lot of similar warnings where the IP in question has the anycast role (so again, assignment to both NICs makes sense). Is there a way around this in the settings.ini file? Netbox itself drops the duplicate IP warning if it detects the anycast role, wondering if the plugin has something analogous.

**Comment by bb-Ricardo** [comment]
Hi,

does your use case apply to AnyCast IP addresses only or are there other types of IP addresses where this also causes issues?

**Comment by Azmodeszer** [comment]
Well, anycast and a few select cases with the VLANs from the opening post. I can live with the latter, since it's no doubt a very uncommon setup, but anycast IPs by their very nature shouldn't trigger the warning, no?

As far as I can see, all other cases are legitimate duplicate IPs that the plugin is right to warn about (our Netbox db still needs some major cleanup).

**Comment by bb-Ricardo** [comment]
For anycast it should definitely be possible. Will need to add a check if IP is defined as an anycast address. Gonna have a look into it.

**Comment by Azmodeszer** [comment]
I'd very much appreciate it. Thank you!

**Comment by Azmodeszer** [comment]
Any progress on this? :)

**Comment by bb-Ricardo** [comment]
Hey @Azmodeszer,

I address the issue regarding anycast IP addresses. Would you be able to checkout the latest `development` branch and see if this resolves the issue?

**Comment by Azmodeszer** [comment]
Thanks. Well, it doesn't complain about duplicates anymore, but is it just ignoring them outright now? It doesn't sync both instances of an anycast IP, but at least it leaves any manually added duplicates alone. Was that the intention?

**Comment by bb-Ricardo** [comment]
Are these anycast IP addresses which the script complains about?

If yes, then I forget to skip the log message. If not, what type and use case do you have in this case exactly again?

**Comment by Azmodeszer** [comment]
I mean, AFAIK the source (vSphere) doesn't specify whether IPs anycast or not, though I could be wrong. The specific example I quoted is technically IPs from two different VLANs with similar IP scopes. I think my post from Dec 12, 2023 was more about the sync ideally importing an IP object anyway if it detects the anycast role for the duplicate on the NetBox side.

**Comment by bb-Ricardo** [comment]
I can't see anything in the vCenter API that would expose an IP address as anycast address.

So to understand it correctly:
* server A has an any cast IP Address assigned.
* the sync discovers a new VM white the same IP
* The existing IP in NetBox is marked as anycast
* A new address should be created?
* This address also will be assigned the anycast attribute?

**Comment by Azmodeszer** [comment]
That would work, as long as the sync doesn't overwrite the existing IP's role. But I think ideally you'd have something in the settings.ini that allows you to toggle whether the import just adds "duplicate" IPs no matter what (and if it does, it should set the IPs to anycast by default, otherwise NB may not allow them in the first place).

**Comment by bb-Ricardo** [comment]
yes, that was the issue In the past. This might have changed in more recent versions. NetBox keeps changing their APIs like other people change their underwear. It's quite a challenge to keep up with the changes and stay compatible.

**Comment by semx** [comment]
@Azmodeszer the anycast part is in #559.

What went wrong: the code recognised the anycast role, logged that such an address can sit on several interfaces, and then skipped the assignment, so the second VM never got the address. It now creates its own object for that interface with the same role and leaves the first one untouched. Same for vip, vrrp, hsrp, glbp and carp, which NetBox treats the same way. It also only understood the role in the shape the API returns, so addresses this tool had created itself were not recognised as anycast on a later run - that is fixed too, and is probably why you still saw warnings after the change in February 2025.

Your quoted warning about the two VLANs is the other half and is untouched: those addresses carry no role, so there is nothing to tell them apart from a real duplicate. Making that work means a config switch that says "create the address anyway". @bb-Ricardo, worth adding, or leave it?
